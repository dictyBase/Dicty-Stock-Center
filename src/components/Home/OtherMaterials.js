// @flow
import React, { Component } from "react"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import { withStyles } from "@material-ui/core/styles"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { fetchInfoPage } from "actions/page"
import styles from "./homeStyles"

const slugName = "dsc-other-materials"

type Props = {
  /** the Auth object taken from the current state */
  auth: Object,
  /** Action creator that fetches data from API */
  fetchInfoPage: Function,
  /** the Page object taken from the current state */
  page: Object,
  /** Checks if data is currently being fetched */
  isFetching: boolean,
  /** Material-UI styling */
  classes: Object,
}

/**
 * OtherMaterials fetches and displays the Other Materials page content.
 */

export class OtherMaterials extends Component<Props> {
  static defaultProps = {
    page: {
      data: {
        attributes: {},
      },
    },
  }
  componentDidMount() {
    this.props.fetchInfoPage(slugName)
  }
  render() {
    const { isFetching, page, classes } = this.props

    if (!isFetching && page.data.attributes.content) {
      return (
        <div className={classes.panelBlue}>
          <InlineEditor
            auth={this.props.auth}
            page={this.props.page}
            slug={slugName}
          />
        </div>
      )
    }
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={6} />
            &nbsp;
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  isFetching: state.page.isFetching,
  page: state.page[slugName],
})

export default connect<*, *, *, *, *, *>(
  mapStateToProps,
  { fetchInfoPage },
)(withStyles(styles)(OtherMaterials))
