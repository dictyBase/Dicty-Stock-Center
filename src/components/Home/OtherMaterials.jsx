// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import Skeleton from 'react-loading-skeleton'
import OtherMaterialsInlineEditor from '../editor/OtherMaterialsInlineEditor'
import { fetchInfoPage } from 'actions/page'
import { Flex, Box } from 'rebass'
import { PanelBlue } from 'styles'

type Props = {
  auth: Object,
  fetchInfoPage: Function,
  page: Object,
  isFetching: boolean
}

class OtherMaterials extends Component<Props> {
  displayName = 'other DSC materials'
  // set defaultprops to prevent console warnings
  static defaultProps = {
    page: {
      data: {
        attributes: {}
      }
    }
  }
  componentDidMount() {
    this.props.fetchInfoPage('dsc-other-materials')
  }
  render() {
    const { isFetching, page } = this.props

    if (!isFetching && page.data.attributes.content) {
      return (
        <PanelBlue>
          <OtherMaterialsInlineEditor
            auth={this.props.auth}
            page={this.props.page}
          />
        </PanelBlue>
      )
    }
    return (
      <Flex justify="center">
        <Box w={'95%'}>
          <Skeleton count={5} />
        </Box>
      </Flex>
    )
  }
}

const mapStateToProps = state => {
  const slugName = 'dsc-other-materials'
  return {
    auth: state.auth,
    isFetching: state.page.isFetching,
    page: state.page[slugName]
  }
}

export default connect(mapStateToProps, { fetchInfoPage })(OtherMaterials)
