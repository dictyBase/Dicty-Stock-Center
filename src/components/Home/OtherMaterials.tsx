import React from "react"
import { useQuery } from "@apollo/react-hooks"
import Grid from "@material-ui/core/Grid"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import InlineEditor from "components/InlineEditor"
import { GET_CONTENT_BY_SLUG } from "queries/queries"
import useStyles from "./homeStyles"

const slugName = "dsc-other-materials"

/**
 * OtherMaterials fetches and displays the OtherMaterials page content.
 */

const OtherMaterials = () => {
  const classes = useStyles()
  const { loading, error, data } = useQuery(GET_CONTENT_BY_SLUG, {
    variables: {
      slug: slugName,
    },
  })

  if (loading) {
    return (
      <Grid container justify="center">
        <Grid item xs={12}>
          <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
            <Skeleton count={6} />
            &nbsp;
          </SkeletonTheme>
        </Grid>
      </Grid>
    )
  }

  if (error) {
    return <div className={classes.panelBlue}>got an error</div>
  }

  return (
    <div className={classes.panelBlue}>
      <InlineEditor data={data.contentBySlug} />
    </div>
  )
}

export default OtherMaterials
