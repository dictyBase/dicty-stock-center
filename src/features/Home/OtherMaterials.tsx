import React from "react"
import { useQuery } from "@apollo/client"
import InlineEditor from "features/EditablePages/InlineEditor"
import PanelLoader from "./PanelLoader"
import { GET_CONTENT_BY_SLUG } from "common/graphql/queries/content"
import useStyles from "./homeStyles"

const slugName = "dsc-other-materials"

/**
 * OtherMaterials fetches and displays the OtherMaterials page content.
 */

const OtherMaterials = () => {
  const classes = useStyles()
  const { loading, error, data } = useQuery(GET_CONTENT_BY_SLUG, {
    variables: {
      slug: slugName,
    },
  })

  if (loading) {
    return <PanelLoader skeletonCount={6} />
  }

  if (error) {
    return (
      <div className={classes.panelBlue}>
        Error fetching other materials information
      </div>
    )
  }

  return (
    <div className={classes.panelBlue}>
      <InlineEditor data={data.contentBySlug} />
    </div>
  )
}

export default OtherMaterials
