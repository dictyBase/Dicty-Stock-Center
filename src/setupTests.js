import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"

global.console = {
  log: jest.fn(),
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
