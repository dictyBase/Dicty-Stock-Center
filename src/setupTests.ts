import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"
import { sprintf } from "sprintf-js"

let globalAny = global as any

globalAny.console = {
  log: (msg: string, args: any) => {
    if (typeof msg !== "string") {
      return
    }
    const str = sprintf(msg, args)
    process.stderr.write(str + "\n")
  },
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
