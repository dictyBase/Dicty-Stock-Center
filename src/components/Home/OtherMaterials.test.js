import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { shallow, mount } from "enzyme"
import sinon from "sinon"
import "../../setupTests"
import { OtherMaterials } from "./OtherMaterials"
import { Flex } from "rebass"

describe("Home/OtherMaterials", () => {
  let props
  let mountedOtherMaterialsPage
  const otherMaterialsPage = () => {
    if (!mountedOtherMaterialsPage) {
      mountedOtherMaterialsPage = mount(<OtherMaterials {...props} />)
    }
    return mountedOtherMaterialsPage
  }

  beforeEach(() => {
    props = {
      page: undefined,
      match: undefined,
      fetchInfoPage: undefined,
      isFetching: undefined,
    }
    mountedOtherMaterialsPage = undefined
  })
  describe("initial render", () => {
    beforeEach(() => {
      props = {
        page: {
          data: {
            attributes: {
              content: "page content",
            },
          },
        },
        fetchInfoPage: () => {},
        isFetching: true,
      }
    })

    it("always renders Flex", () => {
      expect(otherMaterialsPage().find(Flex).length).toBe(1)
    })
    it("calls componentDidMount", () => {
      sinon.spy(OtherMaterials.prototype, "componentDidMount")
      otherMaterialsPage()
      expect(OtherMaterials.prototype.componentDidMount.calledOnce).toEqual(
        true,
      )
    })
  })
})

test("matching a snapshot of OtherMaterials", () => {
  const fetchInfoPage = () => {}
  const page = {
    data: {
      attributes: {},
    },
  }

  const component = renderer.create(
    <OtherMaterials
      page={page}
      fetchInfoPage={fetchInfoPage}
      isFetching={false}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
