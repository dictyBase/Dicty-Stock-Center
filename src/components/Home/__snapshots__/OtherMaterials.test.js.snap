// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of OtherMaterials 1`] = `
<div
  className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-bdVaJa FHiEA"
    is={null}
  >
    <div
      className="react-loading-theme-2"
    >
      <span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
        <span
          className="react-loading-skeleton"
          style={
            Object {
              "animation": "progress 1.2s ease-in-out infinite",
            }
          }
        >
          ‌
        </span>
      </span>
    </div>
  </div>
</div>
`;
